### DISCUSSION TITLE
AES 256GCM for PCIe IDE 96 bit tag

### DISCUSSION THREAD

**Discussion by soumit-r-mitruka** [jedisct1/libsodium#1226] [Q&A] [answered]
Is it possible in its current implementation for the AES GCM 256 to produce a 96 bit MAC and be able to decrypt/verify using a 96 bit MAC also? PCIe IDE implementation is AES GCM 256 and they have chosen a 96 bit MAC instead of the full 128 bits. The receiver of the PCIe packet will only ever receive 96 bits of MAC.

**Comment by jedisct1** [comment, marked as answer]
No.

**Reply by soumit-r-mitruka** [reply]
Is there any plan to support this in future?

**Comment by ivmaykov** [comment]
You will probably have to use OpenSSL for that. I believe it allows using smaller authentication tags for AES-GCM. Good luck.

**Reply by soumit-r-mitruka** [reply]
Thanks for the suggestion. I'll check that out.

**Comment by jedisct1** [comment]
You can still use libsodium to produce these tags (just truncate them to 96 bits), but not to verify them.

**Reply by soumit-r-mitruka** [reply]
Verifying is the prime requirement for the project.
The hardware decrypts the incoming packet with 96 bit MAC hence a need for an independent check to make sure the decryption done by the hardware is correct.
